app/models/webhook/event.rb

```diff
@@ -9,6 +9,8 @@ class Event
     attr_reader :event_type, :payload, :timestamp
 
     def initialize(event_type:, payload: {})
+      raise InvalidEvent unless EVENT_TYPES.include?(event_type)
```

Comment:
I think you can `include ActiveModel::Validations` in the class and use all the validators you're accostumed to for normal model classes: https://api.rubyonrails.org/classes/ActiveModel/Validations.html

Reply:
I think validations are overkill for this case, and I would need to check `valid?` and raise an exception after initializing before dispatching.
I will keep ActiveModel validations in mind if we need more of the validating logic.